content/docs/2.9/scalers/elasticsearch.md

```diff
@@ -42,7 +44,7 @@ triggers:
 
 ### Authentication Parameters
 
-You can authenticate by using a username/password authentication.
+You can authenticate by using a username/password authentication. When using ElasticSearch on Elastic Cloud you can also authenticate by generating an `apiKey` and using your `cloudID` to connect to your cluster.
```

Comment:
Which of these are configurable in TriggerAuthentication? I presume that would be API key?

Reply:
@rcbop PTAL ^

Reply:
Hi @tomkerkhove,

TLDR; I believe both.

Both new parameters are using [GetFromAuthOrMeta()](https://github.com/kedacore/keda/blob/710acf6c2515eb71bb1ac1b4ffaa22c569e725fc/pkg/scalers/scaler.go#L104) helper function function from `keda/pkg/scalers/scaler.go` in [this section](https://github.com/kedacore/keda/pull/3786/files#diff-69370f13dc307d4aa79cc2d0b5787bb37590511753d5dc0e419fbc7b92f69f9eR120-R133) of the ElasticSearch Scaler. I believe this means you could define both `cloudID` or `apiKey`  using TriggerAuthentication, according to [this section in the docs](https://keda.sh/docs/2.8/concepts/authentication/#the-downsides) that would be the recommended way since it decouples secrets from `ScaledObject`. The helper function will try to find the values in the `AuthParams` first and then move into extracting the values from the `TriggerMetadata` fields.

Please correct me if I'm mistaken @JorTurFer.

Best Regards,

cc: @zroubalik

Reply:
That's perfect, can you update the docs to follow our practice on this please?

https://github.com/kedacore/keda-docs#writing-documentation-for-a-scaler

Example: https://keda.sh/docs/2.8/scalers/azure-storage-blob/#authentication-parameters

Reply:
Hi @tomkerkhove, tried to follow your example, please take a look if it's good now.